### DISCUSSION TITLE
Multi Language support ( i18n )

### DISCUSSION THREAD

**Discussion by radicaldrew** [twentyhq/twenty#5709] [Ideas]
I have added i18n support using react-18next including support for RTL languages.
I store language and rtl in local storage and not in user settings.
added language drop down in appearance settings.
If this is of interest to anyone I would be happy to share.

**Comment by eridme** [comment]
Hi! could you please share your solution?

**Reply by radicaldrew** [reply]
yes I can either share it or I can create a a pull request for it.
I have installed react-i18next 
created a i18n folder in modules with the locales and hooks etc.
added a dropdown under packages/twenty-front/src/pages/settings/SettingsAppearance.tsx
to allow for changing the language.
See example module:

```
import { initReactI18next } from 'react-i18next';
import i18n from 'i18next';
import en from './locales/en-US.json';
import he from './locales/he-IL.json';
const resources = {
  en: {
    translation: en,
  },
  he: {
    translation: he,
  },
};

i18n
  .use(initReactI18next)
  .init({
    resources,
    //whitelist: languages,
    //saveMissing: true,
    //loadPath: '/locales/{{lng}}/{{ns}}.json',
    // namespace: 'translation',
    lng: 'en-US',
    interpolation: {
      escapeValue: false, // react already safes from xss
    },
  });

export default i18n;
```

**Reply by laianeramos** [reply]
Could you add pt_BR too, please? @radicaldrew

**Reply by lionelhorn** [reply]
@radicaldrew I'm also interested in how you setup things. 

Did you then go through each component to add `useTranslation` hook?

**Reply by lionelhorn** [reply]
Until twenty supports i18n natively:

For anyone interested I started a POC to transform the codebase by code via a jscode transformer.
In hope to reduce the manual work of going through each file.

I added deepl to auto-translate via their api.

Not all cases are handled but it's a nice starting point
https://github.com/lionelhorn/twenty/blob/i18n/packages/twenty-i18n/README.md

**Comment by LukaszMielczarek** [comment]
Hi. Could you create a PR for me ?? 
I working for a company that would like to use Twenty in their solution, but the lack of multi-language support is a blocker for us. We are willing to contribute and work on this feature but first, we would like to know the current status of this request from the roadmap perspective and if there were any discussions regarding the preferred solution.

**Reply by radicaldrew** [reply]
Yes or I can share the code with you

**Reply by LukaszMielczarek** [reply]
yes, please share a code

**Reply by cruzpeanelo** [reply]
Hello, I have the same requirement. Could share you with me?  PR or code, tks a lot

**Comment by galievruslan** [comment]
Also interested in support for changing the interface language

**Comment by Bonapara** [comment]
We're working on it! 
https://github.com/twentyhq/twenty/pulls?q=is%3Apr+translate

**Comment by FelixMalfait** [comment]
V1 for multi-language has been shipped. It's not perfect, there are still things that need to be improved:
- multi-lingual emails
- translation values for SELECT fields
- translation view names

But it's a good start. I think we can close this, we'll keep improving this.

The source of truth for translation is available here: https://twenty.crowdin.com/twenty ; I think people can ask to join the Crowdin team to contribute/edit translations (but I've never tried it)
